crawler/file_processing.py

```diff
@@ -439,9 +464,49 @@ def insert_samples_from_docs(self, docs_to_insert) -> None:
 
             self.docs_inserted = e.details["nInserted"]
             self.add_duplication_errors(e)
+
+            errored_ids = list(map(lambda x: x['op'][FIELD_MONGODB_ID], e.details["writeErrors"]))
+            inserted_ids = [ doc[FIELD_MONGODB_ID] for doc in docs_to_insert if doc[FIELD_MONGODB_ID] not in errored_ids ]
```

Comment:
Hi @emrojo , we haven't addressed this comment. I'm not sure what other errors we'd expect. This is in a block where we've caught a 'BulkWriteError', so hopefully those errors should be in the 'writeErrors' section?

If there is a different kind of exception thrown, it will get caught in the below block, so we should get to know about it.

You are right, it's possible this would end up causing a discrepancy between the 2 dbs. I'd rather not have to requery the ids from MongoDB though, as I think it would be a v large query statement, which Mongo doesn't like, and would slow down the process.

I'm tempted to leave it, and amend it in future if we become aware of other forms of errors. Do you think it's OK to merge in this state?

Reply:
Yes, I agree with that, is also a type of error we cannot predict, ok to merge